td-agent/Rakefile

```diff
@@ -544,6 +614,70 @@ class BuildTask
     end
   end
 
+  def openssldir
+    File.join(staging_etcdir, "openssl")
+  end
+
+  def create_certs
+    keychains = [
+      "/System/Library/Keychains/SystemRootCertificates.keychain"
+    ]
+
+    cert_list = `security find-certificate -a -p #{keychains.join(" ")}`
+    certs = cert_list.scan(
+      /-----BEGIN CERTIFICATE-----.*?-----END CERTIFICATE-----/m
+    )
+
+    valid_certificates = certs.select do |cert|
+      IO.popen("#{staging_bindir}/openssl x509 -inform pem -checkend 0 -noout > /dev/null", "w") do |ossl_io|
+        ossl_io.write(cert)
+        ossl_io.close_write
+      end
+
+      $?.success?
+    end
+
+    mkdir_p(openssldir)
+    File.open(File.join(openssldir, "cert.pem"), 'w', 0644) do |f|
+      f.write(valid_certificates.join("\n") << "\n")
+    end
+  end
+
+  def build_openssl
+    tarball = @download_task.file_openssl_source
+    source_dir = tarball.sub(/\.tar\.gz$/, '')
+
+    sh("tar", "xvf", tarball, "-C", DOWNLOADS_DIR)
+
+    configure_opts = [
+      "--prefix=#{install_prefix}",
+      "--openssldir=#{File.join("etc", "openssl")}",
+      "no-unit-test",
+      "no-comp",
+      "no-idea",
+      "no-mdc2",
+      "no-rc5",
+      "no-ssl2",
+      "no-ssl3",
+      "no-ssl3-method",
+      "no-zlib",
+      "shared",
+    ]
+    arch_opts = [
+      "darwin64-x86_64-cc",
+    ]
+
+    cd(source_dir) do
+      # Currently, macOS only.
+      sh("perl", "./Configure", *(configure_opts + arch_opts))
+      sh("make", "depend")
+      sh("make")
+      # For `gem install` Ruby.
+      sh("make", "install")
```

Comment:
could you explain a bit futhermore?

Reply:
Ruby which is installed in /opt/td-agent/bin needs to openssl library with the same way.
Same reason for make install execution on https://github.com/fluent-plugins-nursery/td-agent-builder/blob/54366045e988df59ca1e5af7d8822bbb79b72b7d/td-agent/Rakefile#L703.

Reply:
Got it. Thanks.